testsuite/model/src/test/java/org/keycloak/testsuite/model/session/OfflineSessionPersistenceTest.java

```diff
@@ -160,6 +163,32 @@ public void testPersistenceMultipleNodesClientSessionAtSameNode() throws Interru
                 }).forEach(userSessionModel -> clientSessionIds.add(userSessionModel.getId())));
                 return null;
             });
+
+            // ensure that all session have been created on all nodes
+            intermediate.countDown();
+            try {
+                intermediate.await();
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                throw new RuntimeException(e);
+            }
+
+            // defer the shutdown and check if all sessions exist to ensure that they replicate across the different nodes
+            // this should avoid an "org.infinispan.remoting.transport.jgroups.SuspectException: ISPN000400: Node node-XX was suspected"
+            while (true) {
```

Comment:
Rather than indefinite `while`, there should be some timeout / maximum iteration count

Reply:
This is already wrapped in a timeout for the `inIndependentFactories`. I'd rather let this timeout catch this, so that the result shows in the stacktrace where it timed out. This allows me to keep the code simpler. Please confirm.

Reply:
That makes sense, thanks for explanation